lwt-core/cohttp_lwt.ml

```diff
@@ -206,32 +206,29 @@ module Make_client
   let callv ?(ctx=default_ctx) uri reqs =
     Net.connect_uri ~ctx uri >>= fun (conn, ic, oc) ->
     (* Serialise the requests out to the wire *)
-    Lwt_stream.fold_s (fun (req,body) meths ->
+    let meth_stream = Lwt_stream.map_s (fun (req,body) ->
       Request.write (fun writer ->
         Cohttp_lwt_body.write_body (Request.write_body writer) body
       ) req oc >>= fun () ->
-      return ((Request.meth req)::meths)
-    ) reqs [] >>= fun meths ->
+      return (Request.meth req)
+    ) reqs in
     (* Read the responses. For each response, ensure that the previous
        response has consumed the body before continuing to the next
        response because HTTP/1.1-pipelining cannot be interleaved. *)
```

Comment:
This comment seems a little wrong. Does it mean HTTP/1.1 pipelining cannot be *out-of-order*? It is called *pipelining*...